scipy/stats/tests/test_morestats.py

```diff
@@ -522,6 +522,156 @@ def test_result_attributes(self):
+
+    def test_alternative_exact(self):
+        x1 = [-5, 1, 5, 10, 15, 20, 25] # high scale, loc=10
+        x2 = [7.5, 8.5, 9.5, 10.5, 11.5, 12.5] # low scale, loc=10
+        # ratio of scales is greater than 1. So, the
+        # p-value must be high when `alternative='less'`
+        # and low when `alternative='greater'`.
+        pval_l = stats.ansari(x1, x2, alternative='less').pvalue
+        pval_g = stats.ansari(x1, x2, alternative='greater').pvalue
+        assert pval_l > 0.95
+        assert pval_g < 0.05 # level of significance.
+        # sanity check. The result should flip if
+        # we exchange x and y.
+        pval_l_reverse = stats.ansari(x2, x1, alternative='less').pvalue
+        pval_g_reverse = stats.ansari(x2, x1, alternative='greater').pvalue
+        assert pval_l_reverse < 0.05
+        assert pval_g_reverse > 0.95
+
+    @pytest.mark.parametrize(
+        'x, y, alternative, expected',
+        # the tests are designed in such a way that the
+        # if else statement in ansari test for exact
+        # mode is covered.
+        [([1, 2, 3, 4], [5, 6, 7, 8], 'less', 0.6285714285714),
+         ([1, 2, 3, 4], [5, 6, 7, 8], 'greater', 0.6285714285714),
+         ([1, 2, 3], [4, 5, 6, 7, 8], 'less', 0.8928571428571),
+         ([1, 2, 3], [4, 5, 6, 7, 8], 'greater', 0.2857142857143),
+         ([1, 2, 3, 4, 5], [6, 7, 8], 'less', 0.2857142857143),
+         ([1, 2, 3, 4, 5], [6, 7, 8], 'greater', 0.8928571428571)]
+    )
+    def test_alternative_exact_with_R(self, x, y, alternative, expected):
+        # testing with R on random data
+        pval = stats.ansari(x, y, alternative=alternative).pvalue
+        assert_almost_equal(pval, expected)
+
+    def test_alternative_approx(self):
+        # intuitive tests for approximation
+        x1 = stats.norm.rvs(0, 5, size=100, random_state=123)
+        x2 = stats.norm.rvs(0, 2, size=100, random_state=123)
+        # for m > 55 or n > 55, the test should automatically
+        # switch to approximation. So, no need to pass mode='approx'.
+        pval_l = stats.ansari(x1, x2, alternative='less').pvalue
+        pval_g = stats.ansari(x1, x2, alternative='greater').pvalue
+        assert_almost_equal(pval_l, 1.0)
+        assert_almost_equal(pval_g, 0.0)
+
+    @pytest.mark.parametrize(
+        'x, y, alternative, expected',
+        # the tests are designed in such a way that the
+        # if else statement in ansari test for approx
+        # mode is covered.
+        [([1, 2, 3, 4], [5, 6, 7, 8], 'less', 0.5),
+         ([1, 2, 3, 4], [5, 6, 7, 8], 'greater', 0.5),
+         ([1, 2, 3], [4, 5, 6, 7, 8], 'less', 0.8203016145897),
+         ([1, 2, 3], [4, 5, 6, 7, 8], 'greater', 0.1796983854103),
+         ([1, 2, 3, 4, 5], [6, 7, 8], 'less', 0.1796983854103),
+         ([1, 2, 3, 4, 5], [6, 7, 8], 'greater', 0.8203016145897)]
+    )
+    def test_alternative_approx_with_R(self, x, y, alternative, expected):
+        # testing with R on random data
```

Comment:
Same as above. You could just refer to the code above so you don't have to include it again (as long as the example shows how to use approximate vs exact mode).

Reply:
Done.